src/Symfony/Bridge/Twig/EventListener/TemplateAttributeListener.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Twig\EventListener;
+
+use Symfony\Bridge\Twig\Attribute\Template;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\StreamedResponse;
+use Symfony\Component\HttpKernel\Event\ControllerArgumentsEvent;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Twig\Environment;
+
+class TemplateAttributeListener implements EventSubscriberInterface
+{
+    public function __construct(
+        private Environment $twig,
+    ) {
+    }
+
+    public function onKernelView(ViewEvent $event)
+    {
+        $parameters = $event->getControllerResult();
+
+        if (!\is_array($parameters ?? [])) {
+            return;
+        }
+        if (!$attribute = ($event->controllerArgumentsEvent?->getAttributes()[Template::class][0] ?? null)) {
+            return;
+        }
+
+        $parameters ??= $this->resolveParameters($event->controllerArgumentsEvent, $attribute->vars);
```

Comment:
i'd prefer to not do this :)

Reply:
I prefer keeping feature parity as much as possible to ease migration, and also I there is no harm keeping that feature.